tool/tctl/common/notification_command_test.go

```diff
@@ -0,0 +1,113 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/integration/helpers"
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/tool/teleport/testenv"
+)
+
+// TestNotificationCommandCRUD tests creating, listing, and deleting notifications via the `tctl notifications` commands.
+func TestNotificationCommmandCRUD(t *testing.T) {
+	dynAddr := helpers.NewDynamicServiceAddr(t)
+	fileConfig := &config.FileConfig{
+		Global: config.Global{
+			DataDir: t.TempDir(),
+		},
+		Auth: config.Auth{
+			Service: config.Service{
+				EnabledFlag:   "true",
+				ListenAddress: dynAddr.AuthAddr,
+			},
+		},
+	}
+	process := makeAndRunTestAuthServer(t, withFileConfig(fileConfig), withFileDescriptors(dynAddr.Descriptors))
+
+	clt := testenv.MakeDefaultAuthClient(t, process)
+
+	auditorUsername := "auditor-user"
+	managerUsername := "manager-user"
+
+	// Test creating a user-specific notification for auditor user.
+	buf, err := runNotificationsCommand(t, clt, []string{"create", "--user", auditorUsername, "--title", "auditor notification", "--content", "This is a test notification."})
+	require.NoError(t, err)
+	require.Contains(t, buf.String(), "for user auditor-user")
+	auditorUserNotificationId := strings.Split(buf.String(), " ")[2]
+
+	// Test creating a user-specific notification for manager user.
+	buf, err = runNotificationsCommand(t, clt, []string{"create", "--user", managerUsername, "--title", "manager notification", "--content", "This is a test notification."})
+	require.NoError(t, err)
+	require.Contains(t, buf.String(), "for user manager-user")
+
+	// Test creating a global notification for users with the test-1 role.
+	buf, err = runNotificationsCommand(t, clt, []string{"create", "--roles", "test-1", "--title", "test-1 notification", "--content", "This is a test notification."})
+	require.NoError(t, err)
+	require.Contains(t, buf.String(), "for users with one or more of the following roles: [test-1]")
+	globalNotificationId := strings.Split(buf.String(), " ")[2]
+
+	// We periodically check with a timeout since it can take some time for the item to be replicated in the cache and be available for listing.
+	require.EventuallyWithT(t, func(collectT *assert.CollectT) {
+		// List notifications for auditor and verify that auditor notification exists.
+		buf, err = runNotificationsCommand(t, clt, []string{"ls", "--user", auditorUsername})
```

Comment:
If this fails the `t`, then the whole test will fail instead of only failing the `EventuallyWithT`.

I think that's probably fine in this case (making `runNotificationsCommand` work with a `testing.T` or a `assert.CollectT` is probably a bigger change), but wanted to point it out.